aptos-move/aptos-vm/src/transaction_validation.rs

```diff
@@ -48,3 +71,259 @@ impl TransactionValidation {
+}
+
+// Deprecated (see ModuleBundle).
+pub(crate) fn run_module_prologue(
+    session: &mut SessionExt,
+    txn_data: &TransactionMetadata,
+    log_context: &AdapterLogSchema,
+) -> Result<(), VMStatus> {
+    let txn_sequence_number = txn_data.sequence_number();
+    let txn_authentication_key = txn_data.authentication_key();
+    let txn_gas_price = txn_data.gas_unit_price();
+    let txn_max_gas_units = txn_data.max_gas_amount();
+    let txn_expiration_timestamp_secs = txn_data.expiration_timestamp_secs();
+    let chain_id = txn_data.chain_id();
+    let mut gas_meter = UnmeteredGasMeter;
+    session
+        .execute_function_bypass_visibility(
+            &APTOS_TRANSACTION_VALIDATION.module_id(),
+            &APTOS_TRANSACTION_VALIDATION.module_prologue_name,
+            vec![],
+            serialize_values(&vec![
+                MoveValue::Signer(txn_data.sender),
+                MoveValue::U64(txn_sequence_number),
+                MoveValue::vector_u8(txn_authentication_key.to_vec()),
+                MoveValue::U64(txn_gas_price.into()),
+                MoveValue::U64(txn_max_gas_units.into()),
+                MoveValue::U64(txn_expiration_timestamp_secs),
+                MoveValue::U8(chain_id.id()),
+            ]),
+            &mut gas_meter,
+        )
+        .map(|_return_vals| ())
+        .map_err(expect_no_verification_errors)
+        .or_else(|err| convert_prologue_error(err, log_context))
+}
+
+/// Run the prologue for a multisig transaction. This needs to verify that:
+/// 1. The the multisig tx exists
+/// 2. It has received enough approvals to meet the signature threshold of the multisig account
+/// 3. If only the payload hash was stored on chain, the provided payload in execution should
+/// match that hash.
+pub(crate) fn run_multisig_prologue(
+    session: &mut SessionExt,
+    txn_data: &TransactionMetadata,
+    payload: &Multisig,
+    log_context: &AdapterLogSchema,
+) -> Result<(), VMStatus> {
+    let unreachable_error = VMStatus::error(StatusCode::UNREACHABLE, None);
+    let provided_payload = if let Some(payload) = &payload.transaction_payload {
+        bcs::to_bytes(&payload).map_err(|_| unreachable_error.clone())?
+    } else {
+        // Default to empty bytes if payload is not provided.
+        bcs::to_bytes::<Vec<u8>>(&vec![]).map_err(|_| unreachable_error)?
+    };
+
+    session
+        .execute_function_bypass_visibility(
+            &MULTISIG_ACCOUNT_MODULE,
+            VALIDATE_MULTISIG_TRANSACTION,
+            vec![],
+            serialize_values(&vec![
+                MoveValue::Signer(txn_data.sender),
+                MoveValue::Address(payload.multisig_address),
+                MoveValue::vector_u8(provided_payload),
+            ]),
+            &mut UnmeteredGasMeter,
+        )
+        .map(|_return_vals| ())
+        .map_err(expect_no_verification_errors)
+        .or_else(|err| convert_prologue_error(err, log_context))
+}
+
+fn run_epilogue(
```

Comment:
I like that you are moving these our into a separate file, but the epilogue doesn't seem to fall under "transaction validation".

Reply:
Originally, I put it there because we have both prologue and epilogue in `transaction_validation.move`. I will rename "transaction validation" to something else then?

Reply:
Renamed to `transaction_pre_post_processing`